src/shop/migrations/0022_auto_20170313_2018.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.6 on 2017-03-13 20:18
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.utils.timezone
+import django.db.models.deletion
+from django.db.models import Sum
+from django.utils.timezone import timedelta
+import enumfields.fields
+import shop.enums
+import uuid
+import enum
+from moneyed import Money
+
+
+class TrxStatusOld(enum.Enum):
+    FINALIZED = 0
+    CANCELED = 1
+
+
+class TrxStatus(enum.Enum):
+    FINALIZED = 0
+    CANCELED = 1
+    PENDING = 2
+
+
+class TrxType(enum.Enum):
+    FINALIZED = 0
+    PENDING = 1
+    CANCELLATION = 2
+
+
+def forwards_func(apps, schema_editor):
+    # We get the model from the versioned app registry;
+    # if we directly import it, it'll be the wrong version
+    transactions = apps.get_model('shop', 'producttransaction')
+    status_model = apps.get_model('shop', 'producttransactionstatus')
+
+    qs = transactions.objects.all()
+
+    WalletTransaction = apps.get_model("wallet", "WalletTransaction")
+    pre_balance = WalletTransaction.objects \
+        .exclude(trx_type=TrxType.PENDING) \
+        .aggregate(amount=Sum('amount'))['amount'] or Money(0, 'SEK')
+
+    for trx in qs:
+        # Make sure we simulate that any PENDING was created
+        # before FINALIZED
+        status_model.objects.create(
+            date_created=(trx.date_created - timedelta(milliseconds=100)),
+            trx=trx,
+            status=TrxStatus.PENDING,
+            reference_ct=trx.reference_ct,
+            reference_id=trx.reference_id
+        )
+
+        if trx.trx_status.value == TrxStatusOld.FINALIZED.value:
+            status = TrxStatus.FINALIZED
+        else:
+            status = TrxStatus.CANCELED
+
+        status_model.objects.create(
+            date_created=trx.date_created,
+            trx=trx,
+            status=status,
+            reference_ct=trx.reference_ct,
+            reference_id=trx.reference_id
+        )
+
+    post_balance = WalletTransaction.objects \
+        .exclude(trx_type=TrxType.PENDING) \
+        .aggregate(amount=Sum('amount'))['amount'] or Money(0, 'SEK')
+    # Make sure that the total balance in the system has not changed
+    assert pre_balance == post_balance
```

Comment:
This does not seem to be the right type of check here. What we should be checking is that the product quantities remained unchanged after the migration.